### ISSUE TITLE
Feature request: Train test split of CutSet based on supervision ids instead of recordings

### ISSUE THREAD

**Issue by janvainer** [lhotse-speech/lhotse#144]
I have a list of utterance ids that should be in test set or train set. Unfortunately, the train test split may use utterances that share the same recording. To create test set cuts, I use the following:
```
cuts = CutSet.from_yaml("feats/features.yaml")
for cut in cuts:
    cut.supervisions = [supervision for supervision in cut.supervisions if supervision.id in test_ids]
```

Would it make sense to you to provide CutSet api for splitting based on supervision ids? For example:
```
set1, set2, ..., set3 = CutSet.split_on_supervisions(supervision_ids: Iterable[Iterable[str]])
```
This would be used like:
```
train, dev, test = cuts.split_on_supervisions([train_ids, dev_ids, test_ids])
```

**Comment by pzelasko** [comment]
That's a good addition. I'd rather have it be more general by accepting a flat iterable, and then what you need could be achieved by:
```python
train = cuts.subset(supervision_ids=train_ids)
dev = cuts.subset(supervision_ids=dev_ids)
test = cuts.subset(supervision_ids=test_ids)
```

I was thinking we could eventually "mimic" Kaldi `subset_data_dir.sh` a bit, by naming this method as `subset()` and giving it a number of optional args to subset based on different criteria. We could start with what you are suggesting:

```python
def subset(supervision_ids: Optional[Iterable[str]] = None) -> CutSet:
```

Would you like to contribute that?

**Comment by janvainer** [comment]
Hi, your version sounds better. I will look into it over the weekend and make a PR.

**Comment by janvainer** [comment]
Was resolved by #145
